src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiHearingResponseMapper.java

```diff
@@ -59,8 +60,10 @@ public HearingEntity mapHmiHearingErrorToEntity(ErrorDetails hearing, HearingEnt
     public HmcHearingResponse mapEntityToHmcModel(HearingResponseEntity hearingResponseEntity, HearingEntity hearing) {
         HmcHearingResponse hmcHearingResponse = new HmcHearingResponse();
         hmcHearingResponse.setHearingID(hearing.getId().toString());
-        hmcHearingResponse.setCaseRef(hearing.getCaseHearingRequest().getCaseReference());
-        hmcHearingResponse.setHmctsServiceCode(hearing.getCaseHearingRequest().getHmctsServiceID());
+        CaseHearingRequestEntity matchingCaseHearingRequestEntity = hearing
```

Comment:
Updated logic in this class to now use the associated case hearing request based on the response version.